Staged/Classes/UINavigationController+Staged.m

```diff
@@ -0,0 +1,81 @@
+//
+//  UINavigationController+Staged.m
+//  Staged
+//
+//  Created by Gustavo Barbosa on 7/31/16.
+//
+//
+
+#import "UINavigationController+Staged.h"
+#import "UIViewController+StagedSwizziling.h"
+
+NSString * const STGViewControllerWasPushedNotificationName = @"STGViewControllerWasPushedNotificationName";
+NSString * const STGViewControllerWasPoppedNotificationName = @"STGViewControllerWasPoppedNotificationName";
+NSString * const STGNavigationControllerWasPoppedToRootNotificationName = @"STGNavigationControllerWasPoppedToRootNotificationName";
+NSString * const STGNavigationControllerWasPoppedToViewControllerNotificationName = @"STGNavigationControllerWasPoppedToViewControllerNotificationName";
+NSString * const STGNavigationControllerItemsWereSetNotificationName = @"STGNavigationControllerItemsWereSetNotificationName";
+
+NSString * const STGViewControllerPushingAnimatedKey = @"STGViewControllerPushingAnimatedKey";
+NSString * const STGViewControllerPoppingAnimatedKey = @"STGViewControllerPoppingAnimatedKey";
+
+@implementation UINavigationController (Staged)
+
++ (void)stg_swizzle {
+    [self stg_replaceInstanceMethod:@selector(pushViewController:animated:)
+                         withMethod:@selector(stg_pushViewController:animated:)];
+
+    [self stg_replaceInstanceMethod:@selector(popViewControllerAnimated:)
+                         withMethod:@selector(stg_popViewControllerAnimated:)];
+
+    [self stg_replaceInstanceMethod:@selector(popToRootViewControllerAnimated:)
+                         withMethod:@selector(stg_popToRootViewControllerAnimated:)];
+
+    [self stg_replaceInstanceMethod:@selector(popToViewController:animated:)
+                         withMethod:@selector(stg_popToViewController:animated:)];
+
+    [self stg_replaceInstanceMethod:@selector(setViewControllers:animated:)
+                         withMethod:@selector(stg_setViewControllers:animated:)];
+}
+
+- (void)stg_pushViewController:(UIViewController *)viewControllerToPush
+                      animated:(BOOL)flag {
+    NSNotificationCenter *nc = [NSNotificationCenter defaultCenter];
+    [nc postNotificationName:STGViewControllerWasPushedNotificationName
+                      object:viewControllerToPush
+                    userInfo:@{STGViewControllerPushingAnimatedKey: @(flag)}];
+}
+
+- (nullable UIViewController *)stg_popViewControllerAnimated:(BOOL)flag {
+    NSNotificationCenter *nc = [NSNotificationCenter defaultCenter];
+    [nc postNotificationName:STGViewControllerWasPoppedNotificationName
+                      object:self.topViewController
+                    userInfo:@{STGViewControllerPoppingAnimatedKey: @(flag)}];
+    return nil;
```

Comment:
shouldn't these methods return the view controller that would be popped? It's not a big issue since I've never seen any code using the return code, but who knows ¯\_(ツ)_/¯